test/util/e2e.go

```diff
@@ -377,6 +377,7 @@ func RestartKueueController(ctx context.Context, k8sClient client.Client, kindCl
 	WaitForKueueAvailabilityNoRestartCountCheck(ctx, k8sClient)
 	waitForDeploymentWithOnlyAvailableReplicas(ctx, k8sClient, kcmKey)
 	WaitForLeaderElection(ctx, k8sClient, restartStartTime)
+	WaitForWebhookReady(ctx, k8sClient)
```

Comment:
Hm, I can see this can help our tests, but generally we wait here for "KueueAvailability" and this should already wait for "Ready" pods. And Kueue's Pods should be marked as "Ready" only when certs are available, but maybe we should also check somehow if webhooks ready inside the probes?

Reply:
As discussed over slack, the issue isn't about pod readiness or webhook/cert availability, those are already covered by the existing probes. The problem is the `EndpointSlice` propagation delay: there's a gap between when a pod becomes Ready and when the `EndpointSlice` controller updates the service endpoints with the new pod's IP.